.circleci/config.yml

```diff
@@ -26,19 +26,49 @@ jobs:
           venv/bin/pip install --progress-bar off --upgrade -r requirements.txt
     - run:
         name: PyTest
-        command: venv/bin/pytest --black --flake8
+        command: venv/bin/pytest --black --flake8 --ignore=pensieve/tests/integration/
     - run:
         name: Mypy
         command: venv/bin/mypy pensieve
     - save_cache:
         paths:
         - venv/
         key: *cache_key
+  integration:
+    docker:
+    - image: python:3.8
+    steps:
+    - checkout
+    - &skip_forked_pr
+      run:
+        name: Early return if this build is from a forked PR
+        command: |
+          if [ -n "$CIRCLE_PR_NUMBER" ]; then
+            echo "Cannot pass creds to forked PRs, so marking this step successful"
+            circleci step halt
+          fi
+    - &build
+      run:
+        name: Build
+        command: |
+          python3.8 -m venv venv/
+          venv/bin/pip install --progress-bar off --upgrade -r requirements.txt
+    - &pytest_integration_test
+      run:
+        name: PyTest Integration Test
+        # Google's client libraries will check for GOOGLE_APPLICATION_CREDENTIALS
+        # and use a file in that location for credentials if present;
+        # See https://cloud.google.com/docs/authentication/production
+        command: |
+          export GOOGLE_APPLICATION_CREDENTIALS="/tmp/gcp.json"
```

Comment:
I'm not sure I follow why the location is `/tmp/gcp.json`. Is `GOOGLE_APPLICATION_CREDENTIALS` set to that location in the circleci environment?

Reply:
In circleci `GCLOUD_SERVICE_KEY` is set to the service account key JSON. `GOOGLE_APPLICATION_CREDENTIALS` has to be a file path. Here we just create a temporary file `/tmp/gcp.json` and pipe the `GCLOUD_SERVICE_KEY` content into the file. That way `GOOGLE_APPLICATION_CREDENTIALS` can get picked up correctly when pensieve is run in the CI. It's a bit of a hack.